packages/multi-test/src/wasm.rs

```diff
@@ -78,38 +80,42 @@ where
     fn contract_data(&self, storage: &dyn Storage, address: &Addr) -> AnyResult<ContractData>;
 
     /// Admin interface, cannot be called via CosmosMsg
-    fn sudo(
+    fn sudo<BankT, CustomT>(
         &self,
         api: &dyn Api,
         contract_addr: Addr,
         storage: &mut dyn Storage,
-        router: &Router<ExecC, QueryC>,
+        router: &Router<BankT, CustomT, Self>,
         block: &BlockInfo,
         msg: Vec<u8>,
-    ) -> AnyResult<AppResponse>;
+    ) -> AnyResult<AppResponse>
+    where
+        Self: Sized,
+        BankT: Bank,
+        CustomT: CustomHandler<ExecC = ExecC, QueryC = QueryC>;
 }
 
-pub struct WasmKeeper<ExecC = Empty, QueryC = Empty> {
+pub struct WasmKeeper<ExecC, QueryC> {
     /// code is in-memory lookup that stands in for wasm code
     /// this can only be edited on the WasmRouter, and just read in caches
     codes: HashMap<usize, Box<dyn Contract<ExecC>>>,
-    /// Just marker to make type elision fork when using it as `Wasm` trait
-    _q: std::marker::PhantomData<QueryC>,
+    /// Just markers to make type elision fork when using it as `Wasm` trait
+    _p: std::marker::PhantomData<(ExecC, QueryC)>,
```

Comment:
Rust question:

Why we need them both here when `ExecC` is fixed by codes type already?

Reply:
We don't overlook (I changed this type like all the times in the universe).